hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -281,7 +318,19 @@ private boolean writeUpdateRecord(HoodieRecord<T> hoodieRecord, GenericRecord ol
         return false;
       }
     }
-    return writeRecord(hoodieRecord, indexedRecord, isDelete);
+    boolean result = writeRecord(hoodieRecord, indexedRecord, isDelete);
+    if (cdcEnabled) {
+      String recordKey = StringUtils.join(
```

Comment:
This might not be matching the record key. We should be using proper record-key in here.

Reply:
this is a custom key just for cdc when there is more that one record key which users provide.
and this key will be parsed and used in `HoodieCDCRDD`.